packages/terra-list/src/List.jsx

```diff
@@ -58,10 +58,20 @@ const List = ({
   if (role && role.length > 0 && role !== 'none') {
     attrSpread.role = role;
   }
-
+  let allItems = children ? [] : undefined;
+  if (role === 'listbox') {
+    React.Children.forEach(children, (child) => {
```

Comment:
We don't want to loop through children and clone elements here. The goal is to have to consuming HOC set all relevant data on the list and list items. This reduces the impact of large lists. If a list item needs a prop set it should only the first time by the consumer.

Reply:
Added Prop in df874d5